server/chat-commands/admin.ts

```diff
@@ -824,78 +882,36 @@ export const commands: ChatCommands = {
 		if (Monitor.updateServerLock) {
 			return this.errorReply(`/updateserver - Another update is already in progress (or a previous update crashed).`);
 		}
-		if (isPrivate && (!Config.privatecodepath || !path.isAbsolute(Config.privatecodepath))) {
-			return this.errorReply("`Config.privatecodepath` must be set to an absolute path before using /updateserver private.");
-		}
 
+		const validPrivateCodePath = Config.privatecodepath && path.isAbsolute(Config.privatecodepath);
+		target = toID(target);
 		Monitor.updateServerLock = true;
 
-		const exec = (command: string) => bash(command, this, isPrivate ? Config.privatecodepath : undefined);
-
-		this.addGlobalModAction(`${user.name} used /updateserver${isPrivate ? ` private` : ``}`);
-		this.sendReply(`Fetching newest version...`);
-
-		let [code, stdout, stderr] = await exec(`git fetch`);
-		if (code) throw new Error(`updateserver: Crash while fetching - make sure this is a Git repository`);
-		if (!stdout && !stderr) {
-			this.sendReply(`There were no updates.`);
-			Monitor.updateServerLock = false;
-			return;
-		}
-
-		[code, stdout, stderr] = await exec(`git rev-parse HEAD`);
-		if (code || stderr) throw new Error(`updateserver: Crash while grabbing hash`);
-		const oldHash = String(stdout).trim();
-
-		[code, stdout, stderr] = await exec(`git stash save "PS /updateserver autostash"`);
-		let stashedChanges = true;
-		if (code) throw new Error(`updateserver: Crash while stashing`);
-		if ((stdout + stderr).includes("No local changes")) {
-			stashedChanges = false;
-		} else if (stderr) {
-			throw new Error(`updateserver: Crash while stashing`);
+		if (target === 'private') {
+			if (!validPrivateCodePath) {
+				throw new Chat.ErrorMessage("`Config.privatecodepath` must be set to an absolute path before using /updateserver private.");
+			}
+			await updateserver(this, Config.privatecodepath);
+			this.addGlobalModAction(`${user.name} used /updateserver private`);
 		} else {
-			this.sendReply(`Saving changes...`);
+			if (validPrivateCodePath) await updateserver(this, Config.privatecodepath);
+			await updateserver(this, path.resolve(`${__dirname}/../..`));
+			this.addGlobalModAction(`${user.name} used /updateserver`);
 		}
 
-		// errors can occur while rebasing or popping the stash; make sure to recover
-		try {
-			this.sendReply(`Rebasing...`);
-			[code] = await exec(`git rebase --no-autostash FETCH_HEAD`);
-			if (code) {
-				// conflict while rebasing
-				await exec(`git rebase --abort`);
-				throw new Error(`restore`);
-			}
-
-			if (stashedChanges) {
-				this.sendReply(`Restoring saved changes...`);
-				[code] = await exec(`git stash pop`);
-				if (code) {
-					// conflict while popping stash
-					await exec(`git reset HEAD .`);
-					await exec(`git checkout .`);
-					throw new Error(`restore`);
-				}
-			}
+		this.sendReply(`Rebuilding...`);
+		await rebuild(this);
+		this.sendReply(`DONE: code is up to date and rebuilt!`);
```

Comment:
```suggestion
		this.sendReply(`DONE`);
```

I know the old message wasn't as terse, but I think I do prefer it this way; I don't think it says anything important that wasn't already said (especially since it doesn't even mention if it updated public/private/both).

Reply:
I think anyone who has the permissions to use this command on Main can just read the code anyway, so I don't really care how terse it is.